src/shared/user-helpers.ts

```diff
@@ -1,13 +1,58 @@
 import { UserProfile } from "@auth0/nextjs-auth0";
+import md5 from "md5";
+import querystring from "query-string";
 
 import { User } from "../types";
 
+export interface GravatarUrlProps {
+  email?: string | null;
+  size?: number;
+  rating?: string;
+  theme?: string;
+  protocol?: string;
+  domain?: string;
+}
+const defaultGravatarUrlProps = {
+  email: undefined,
+  size: 50,
+  rating: "g",
+  theme: "mp",
+  protocol: "//",
+  domain: "www.gravatar.com",
+};
+
+export const getGravatarUrlByEmail = (customProps: GravatarUrlProps) => {
```

Comment:
Keeping our own implementation means we’ll need to update multiple dependencies and ideally have some unit tests long-term. Perhaps, it’s cheaper to replace it with a dependency on https://www.npmjs.com/package/gravatar-url or [some similar library](https://www.npmjs.com/search?q=gravatar). They’ll do the maintenance for us.

If you prefer a local implementation, let’s add a comment with a link to docs. Otherwise, it is not easy to validate the logic. We can also explain why we don’t use an off-the-shelf solution for generating gravatars, because that can be a common question amongst collaborators.

Reply:
Thanx for the link to the package, I'll update my RP.